lib/media/drm_engine.js

```diff
@@ -752,6 +752,16 @@ shaka.media.DrmEngine = class {
           return;
         }
       }
+      let duplicate = false;
+      this.storedPersistentSessions_.forEach((metadata, sessionId) => {
+        if (shaka.util.BufferUtils.equal(initData, metadata.initData)) {
+          duplicate = true;
+        }
+      });
```

Comment:
Comparing buffers might be expensive, can you modify this loop to break as soon as we find a duplicate?

Reply:
Done!